src/oracle/middleware.js

```diff
@@ -0,0 +1,805 @@
+
+// get environment variables
+const {
+  PORT = '3000',
+  EI_CHAINLINKURL,
+  FROM,
+  SUBMIT_RETRIES = 3,
+  BLOCK_INTERVAL = '6',
+  SEND_CHECK_INTERVAL = '12',
+  AGORIC_RPC = "http://0.0.0.0:26657",
+  STATE_FILE = "data/middleware_state.json",
+  CREDENTIALS_FILE = "config/ei_credentials.json",
+  OFFERS_FILE = "config/offers.json",
+  FEEDS_FILE = "oracle/feeds.json"
+} = process.env;
+
+/** 
+  * Environment variables validation
+  */
+if (process.env.NODE_ENV != "test"){
+  assert(EI_CHAINLINKURL, '$EI_CHAINLINKURL is required');
+  assert(Number(SUBMIT_RETRIES), '$SUBMIT_RETRIES is required');
+  assert(FROM, '$FROM is required');
+  assert(validUrl(AGORIC_RPC), '$AGORIC_RPC is required');
+  assert(STATE_FILE != "", '$STATE_FILE is required');
+  assert(CREDENTIALS_FILE != "", '$CREDENTIALS_FILE is required');
+  assert(OFFERS_FILE != "", '$OFFERS_FILE is required');
+  assert(FEEDS_FILE != "", '$FEEDS_FILE is required');
+}
+
+var agoricNames = {};
+var fromBoard = {};
+var vstorage = {};
+const marshaller = boardSlottingMarshaller();
+
+var credentials;
+var feeds;
+
+/**
+  * Function to load from file or initialise it
+  * @returns empty state or state from file
+  */
+const readState = () => {
+  // try to read JSON file and if it doesnt exist, create one
+  try {
+    //read JSON file
+    return readJSONFile(STATE_FILE)
+  } catch (err) {
+
+    //create an initial state
+    let initial_state = {
+      "jobs": [],
+      "previous_results": {}
+    }
+
+    //save state
+    saveJSONDataToFile(initial_state, STATE_FILE)
+    return initial_state
+  }
+}
+
+/**
+  * Function to initialise state
+  */
+const initialiseState = () => {
+  //read state
+  let state = readState()
+
+  //go through each job
+  for (let index in state.jobs) {
+    let currentJob = state.jobs[index]
+    //if not initialised
+    if (!(currentJob.name in state.previous_results)) {
+      state.previous_results[currentJob.name] = {
+        id: currentJob.job,
+        result: 0,
+        round: {}
+      }
+    }
+  }
+
+  //save state
+  saveJSONDataToFile(state, STATE_FILE);
+}
+
+/**
+  * Function to get the job's index
+  * @param {*} jobName the job name 
+  * @returns the index of the job
+  */
+const getJobIndex = (jobName) => {
+  //read state
+  let state = readState()
+
+  //go through each job
+  for (let index in state.jobs) {
+    let currentJob = state.jobs[index]
+    //if not initialised
+    if (currentJob.name == jobName) {
+      return index
+    }
+  }
+  return -1
+}
+
+
+/**
+  * Function to send a job run to the CL node
+  * @param {*} credentials the external initiator credentials
+  * @param {*} count the request id
+  * @param {*} jobId the Chainlink external job id
+  * @param {*} chainlinkUrl the Chainlink node url where to send the job request
+  * @param {*} requestType the request type, 1 = time, 2 = deviation, 3 = new round
+  */
+const sendJobRun = async (credentials, count, jobId, chainlinkUrl, requestType) => {
```

Comment:
Doesn't fit 80 char width limit.

```suggestion
const sendJobRun = async (
    credentials, count, jobId, chainlinkUrl, requestType
) => {
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/75621d3c2bf7c76a40ad33451c594eb424529c8e